### ISSUE TITLE
randomIO issue

### ISSUE THREAD

**Issue by yaitskov** [snapframework/snap-core#298]
Hi,

I generated a template and cabal cannot build it.
I also tried ghc 8.8.3.
 
```
$ snap init barebones
$ snap 
$ cabal install
Wrote tarball sdist to
/home/dan/demo/haskell/snap/snap1/dist-newstyle/sdist/snap1-0.1.tar.gz
Resolving dependencies...
Build profile: -w ghc-8.6.5 -O1
In order, the following will be built (use -v for more details):
 - snap-core-1.0.4.1 (lib) (requires build)
 - snap-server-1.1.1.2 (lib) (requires build)
 - snap1-0.1 (exe:snap1) (requires build)
Starting     snap-core-1.0.4.1 (lib)
Building     snap-core-1.0.4.1 (lib)

Failed to build snap-core-1.0.4.1.
Build log (
/home/dan/.cabal/logs/ghc-8.6.5/snap-core-1.0.4.1-303f8a55d6cf492ec512aa944592452970271ab958c6065185caddc5cfeba31c.log
):
Configuring library for snap-core-1.0.4.1..
Preprocessing library for snap-core-1.0.4.1..
Building library for snap-core-1.0.4.1..
[ 1 of 18] Compiling Snap.Internal.Debug ( src/Snap/Internal/Debug.hs, dist/build/Snap/Internal/Debug.o )
[ 2 of 18] Compiling Snap.Types.Headers ( src/Snap/Types/Headers.hs, dist/build/Snap/Types/Headers.o )
[ 3 of 18] Compiling Snap.Internal.Http.Types ( src/Snap/Internal/Http/Types.hs, dist/build/Snap/Internal/Http/Types.o )
[ 4 of 18] Compiling Snap.Internal.Test.Assertions ( src/Snap/Internal/Test/Assertions.hs, dist/build/Snap/Internal/Test/Assertions.o )
[ 5 of 18] Compiling Snap.Internal.Parsing ( src/Snap/Internal/Parsing.hs, dist/build/Snap/Internal/Parsing.o )
[ 6 of 18] Compiling Snap.Internal.Core ( src/Snap/Internal/Core.hs, dist/build/Snap/Internal/Core.o )
[ 7 of 18] Compiling Snap.Internal.Routing ( src/Snap/Internal/Routing.hs, dist/build/Snap/Internal/Routing.o )
[ 8 of 18] Compiling Snap.Internal.Instances ( src/Snap/Internal/Instances.hs, dist/build/Snap/Internal/Instances.o )
[ 9 of 18] Compiling Snap.Core        ( src/Snap/Core.hs, dist/build/Snap/Core.o )
[10 of 18] Compiling Snap.Internal.Util.FileUploads ( src/Snap/Internal/Util/FileUploads.hs, dist/build/Snap/Internal/Util/FileUploads.o )
[11 of 18] Compiling Snap.Internal.Util.FileServe ( src/Snap/Internal/Util/FileServe.hs, dist/build/Snap/Internal/Util/FileServe.o )
[12 of 18] Compiling Snap.Internal.Test.RequestBuilder ( src/Snap/Internal/Test/RequestBuilder.hs, dist/build/Snap/Internal/Test/RequestBuilder.o )

src/Snap/Internal/Test/RequestBuilder.hs:62:46: error:
    Module ‘System.Random’ does not export ‘Random(randomIO)’
   |
62 | import           System.Random              (Random (randomIO))
   |                                              ^^^^^^^^^^^^^^^^^
cabal: Failed to build snap-core-1.0.4.1 (which is required by snap1-0.1). See
the build log above for details.

```

**Comment by yaitskov** [comment]
I guess the issue is cause by old snap (1.0.0).

**Comment by yaitskov** [comment]
it is fixed in trunk but not released. Latest release 1.0.4.1 is not buildable with 8.6.5 or 8.8.3.

**Comment by yaitskov** [comment]
my workaround is cabal.project with

```
source-repository-package
  type: git
  location: file:///home/dan/demo/haskell/snap/snap-core
  commit: 328963b093e60e82ac1317d0476bcf5e8d35843f

packages: .
```

**Comment by tmbdev** [comment]
I'm trying to install ImplicitCAD and this bug in snap-core breaks the install.

My point is: this bug is breaking unrelated end-user software installs, like "cabal install implicit", and there is no obvious workaround for end users.

It would seem rather critical to fix this, even if just by reverting to the previous version.

**Comment by tom-audm** [comment]
@mightybyte we're hitting this in prod as well; can you cut a release with this fix?

**Comment by mightybyte** [comment]
Released as 1.0.4.2.  Thanks.
